packages/context/src/binding.ts

```diff
@@ -52,24 +124,66 @@ export class Binding {
     return undefined;
   }
 
-  private readonly _key: string;
-  private _tags: Set<string> = new Set();
+  public readonly key: string;
+  public readonly tags: Set<string> = new Set();
+  public scope: BindingScope;
+
+  private _cache: BoundValue;
+  private _getValue: (ctx?: Context) => BoundValue | Promise<BoundValue>;
 
   // For bindings bound via toClass, this property contains the constructor
   // function
   public valueConstructor: Constructor<BoundValue>;
 
-  constructor(_key: string, public isLocked: boolean = false) {
-    Binding.validateKey(_key);
-    this._key = _key;
+  constructor(key: string, public isLocked: boolean = false,
+    scope: BindingScope = BindingScope.TRANSIENT) {
+    Binding.validateKey(key);
+    this.key = key;
+    this.scope = scope;
```

Comment:
_I think you missed one of my comments related to this constructor function - https://github.com/strongloop/loopback-next/pull/385#discussion_r123497687._

FWIW, I think this line can be omitted and the property can be declared via a constructor argument:

```ts
private _scope: BindingScope = BindingScope.TRANSIENT;
```

Also I think the constructor has too many parameters at this point, we should stop adding more parameters and provide fluent API for setting these extra flags instead (I see you already have `inScope` method for that).

I am proposing to remove `scope` argument from the constructor completely.